### PR TITLE
save hyperparams to allow clean multi-class model reload, closes #287

### LINKED ISSUE #287: Improve multi-class model reloading
Loading a multi-class model requires you to know the label dict and preset the classes, this is cumbersome.

If you use main.save_model and are loading a multi_class model, it requires you to create a new object, since main.deepforest() instantiates a deepforest single class model by default.

```
TRAINED_MODEL = "/blue/ewhite/everglades/Zooniverse/20211215_112228/species_model.pl"
train = pd.read_csv("/blue/ewhite/everglades/Zooniverse/parsed_images/species_train.csv")
label_dict = {key:value for value, key in enumerate(train.label.unique())}
m = main.deepforest(num_classes=len(label_dict), label_dict=label_dict)
m.load_state_dict(torch.load(TRAINED_MODEL, map_location="cpu")["state_dict"])
```

In the mean time the docs need to be clearer.

### PR DESCRIPTION
**Pull request by bw4sz**
If this passes we should bumpversion patch on main to trigger pypi rebuild.

### CODE REVIEW

**Comment by bw4sz** [comment]
I believe this test is sufficient 

```
def test_reload_multi_class(two_class_m, tmpdir):
    two_class_m.config["train"]["fast_dev_run"] = True
    two_class_m.create_trainer()
    two_class_m.trainer.fit(two_class_m)
    two_class_m.save_model("{}/checkpoint.pl".format(tmpdir))
    before = two_class_m.trainer.validate(two_class_m)
    
    #reload
    old_model = main.deepforest.load_from_checkpoint("{}/checkpoint.pl".format(tmpdir))
    old_model.config = two_class_m.config
    assert old_model.num_classes == 2
    old_model.create_trainer()    
    after = old_model.trainer.validate(old_model)
    
    assert after[0]["val_classification"] == before[0]["val_classification"]
```

If so we should add relevant docs.
